<?xml version="1.0"?>
<launch>

    <arg name="start_pt" default= '14.47, 1.60, 0.00, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/gimp/floor2_rot.yaml" />
    <arg name="run_lidar" default="true"/>

    <group if="$(eval is_sim == true and use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2_rot.rviz" />
    </group>

    <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
    <rosparam command="load" file="$(find svea_core)/resources/params/obstacles_rot.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/track.yaml" />

    <node pkg="svea_core" type="floor2_mpc_co_av.py" name="floor2_mpc_co_av" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="run_lidar" value="$(arg run_lidar)"/>
    </node>

    <node pkg="svea_core" type="map_explore_node.py" name="explore_node" output="screen" />
    <node pkg="svea_core" type="collision_node.py" name="collision_node" output="screen" />
    <node pkg="svea_core" type="global_path_node.py" name="global_path_node" output="screen" />    

</launch>
